interfaces/dotnet/Cantera/src/ThermoPhase.cs

```diff
@@ -55,8 +54,8 @@ static ThermoPhase()
 
     internal ThermoPhase(string filename, string? phaseName)
     {
-        _sol = LibCantera.soln_newSolution(filename, phaseName ?? "", "none");
-        _handle = LibCantera.soln_thermo(_sol);
+        _sol = LibCantera.sol3_newSolution(filename, phaseName ?? "", "none");
+        _handle = LibCantera.sol3_thermo(_sol);
```

Comment:
Until we add a proper `Solution` class, we only need to temporarily create the solution object on the native side (it also appears that it would be safe to destroy it once we extract the thermo object because it uses shared pointers). And to ensure the native solution object is released, we should dispose the handle by leveraging the `using` statement (also guaranteeing its release if an exception is thrown later in the constructor). Also, we need to call `sol.EnsureValid()` before using the handle to get the proper error message in case of a missing file.

```suggestion
        using var sol = LibCantera.sol3_newSolution(filename, phaseName ?? "", "none");
        sol.EnsureValid();
        _handle = LibCantera.sol3_thermo(sol);
``` 

(this assumes we remove the `_sol` field above)

Reply:
See above: I'm not sufficiently familiar - this may introduce side effects that go beyond my wheelhouse.

Reply:
@ischoegl I realized the handle handling is messed up in a number of ways. For example, right now when a `ThermoPhaseHandle` is disposed in .NET it calls `thermo3_del(handle)`. But `thermo3_del` takes no arguments! I am surprised this doesn’t crash. Same problem for `KineticsHandle` and `TransportHandle`.

So I think we should merge this now, and I’ll create a follow-up to fix the these handles. In truth `ThermoPhaseHandle`, `KineticsHandle`, and `TransportHandle`  shouldn’t be treated as such because they don’t identify an owned resource which can be closed independently. It will require some thinking on how the handle classes are scaffolded and which ones to scaffold at all, and how the wrapper classes are scaffolded.